src/message_dao.py

```diff
@@ -12,27 +12,94 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
+import subprocess
+
 from typing import Dict, Optional
+
+import pymysql
+import pickle
 from message import Message
+from google.cloud.sql.connector import Connector
+from dotenv import load_dotenv
 
 EPOCH_HASH = 'ae9e7be4a03765456fe38287533e6446e8bbc93c'
 
 class MessageDao(object):
 
+    connector = Connector()
+
+    load_dotenv()
+    connection = connector.connect(
+        os.environ.get("HOST"),
+        "pymysql",
+        user = os.environ.get("USER"),
+        password = os.environ.get("PASSWORD")
+        )
+
     def __init__(self) -> None:
-        # Maps message.id to message
-        self._messages_seen : Dict[str, Message] = {}
+        db_name = os.environ.get("DB")
+        if not db_name:
+            raise Exception("Missing environment variable for name of database.")
+        with self.connection.cursor() as cursor:
+            cursor.execute("CREATE DATABASE IF NOT EXISTS " + db_name)
+            self.connection.select_db(db_name)
+            # Mapping from message id to message
+            cursor.execute(
+                "CREATE TABLE IF NOT EXISTS Messages"
+                "(message_id VARCHAR(255) NOT NULL,"
+                "binary_message MEDIUMBLOB NOT NULL,"
+                "PRIMARY KEY (message_id))"
+            )
+            # Mapping from name of state attribute to 
+            cursor.execute(
+                "CREATE TABLE IF NOT EXISTS States"
+                "(state_name VARCHAR(255) NOT NULL,"
+                "value VARCHAR(255) NOT NULL,"
+                "PRIMARY KEY (state_name))"
+            )
+        self.connection.commit()
+        self.local_mapping = self._get_all()
```

Comment:
I talked with Lenhard about this offline.

The idea was to not commit messages to the DB until after we uploaded them.
And so to make the `find()` function work as before, we needed some way of tracking "found-but-not-yet-uploaded" messages.

I suggested that we get rid of `find()` entirely, eliminating the need for `local_mapping`.
The places that use it can just use the Message objects they already had access to instead of going through the archive object.